cookbook/sm-s0301.rst

```diff
@@ -0,0 +1,216 @@
+SM-S0301 WiFi Smart Power Strip with USB
+========================================
+
+The SM-S0301 is a multi-outlet power strip with four individually controllable
+NEMA 5-15R receptacles and four USB Type A connectors that are controlled as a
+group.
+
+There is a single button surrounded by blue and red LEDs. There are red LEDs
+next to each socket but they are not controllable from software.
+
+Based upon the Tuya platform, it is rebranded and resold by many companies.
+Installing ESPHome with Tuya-Convert should be possible. If not, it is
+relatively easy to remove the rear cover and access the ESP8266 module,
+although some soldering is necessary to connect wires to the ESP8266 module.
+
+See the Blakaddr_ Tasmota template respository for pictures and reflashing
+information.
+
+.. code-block:: yaml
+
+    substitutions:
+      device_name: sms0301_multi_outlet_power_strip
+      friendly_name: SM-S0301 Multi-Outlet Power Strip
+
+    esphome:
+      name: ${device_name}
+      platform: ESP8266
+      board: esp01_1m
+
+    wifi:
+      ssid: !secret wifi_ssid
+      password: !secret wifi_password
+
+    api:
+      password: !secret api_password
+      reboot_timeout: 0s
+
+    ota:
+      password: !secret ota_password
+
+    logger:
+
+    status_led:
+      pin:
+        number: GPIO0
+        inverted: true
+
+    sensor:
+      - platform: uptime
+        name: ${friendly_name} Uptime Sensor
+
+      - platform: wifi_signal
+        name: ${friendly_name} WiFi Signal Sensor
+        update_interval: 60s
+
+    binary_sensor:
+      - platform: status
+        name: ${friendly_name} Status
+
+      - platform: gpio
+        id: power_button
+        name: ${friendly_name} Power Button
+        internal: true
+        pin:
+          number: GPIO5
+          mode: INPUT_PULLUP
+        on_press:
+          then:
+            - if:
```

Comment:
Add a comment above this explaining what this automation does:

 - If all relays are off, turn them all on.
 - If any relay is on, turn them all off.